File: msgraph-mail/go/utilities/models/inference_classification_override.go
package models

import (
    i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f "github.com/microsoft/kiota-abstractions-go"
    i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91 "github.com/microsoft/kiota-abstractions-go/serialization"
)

// InferenceClassificationOverride 
type InferenceClassificationOverride struct {
    Entity
    // The classifyAs property
    classifyAs *InferenceClassificationType
    // The senderEmailAddress property
    senderEmailAddress EmailAddressable
}
// NewInferenceClassificationOverride instantiates a new InferenceClassificationOverride and sets the default values.
func NewInferenceClassificationOverride()(*InferenceClassificationOverride) {
    m := &InferenceClassificationOverride{
        Entity: *NewEntity(),
    }
    return m
}
// CreateInferenceClassificationOverrideFromDiscriminatorValue creates a new instance of the appropriate class based on discriminator value
func CreateInferenceClassificationOverrideFromDiscriminatorValue(parseNode i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode)(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.Parsable, error) {
    return NewInferenceClassificationOverride(), nil
}
// GetClassifyAs gets the classifyAs property value. The classifyAs property
func (m *InferenceClassificationOverride) GetClassifyAs()(*InferenceClassificationType) {
    return m.classifyAs
}
// GetFieldDeserializers the deserialization information for the current model
func (m *InferenceClassificationOverride) GetFieldDeserializers()(map[string]func(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode)(error)) {
    res := m.Entity.GetFieldDeserializers()
    res["classifyAs"] = i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.SetEnumValue(ParseInferenceClassificationType , m.SetClassifyAs)
    res["senderEmailAddress"] = i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.SetObjectValue(CreateEmailAddressFromDiscriminatorValue , m.SetSenderEmailAddress)
    return res
}
// GetSenderEmailAddress gets the senderEmailAddress property value. The senderEmailAddress property
func (m *InferenceClassificationOverride) GetSenderEmailAddress()(EmailAddressable) {
    return m.senderEmailAddress
}
// Serialize serializes information the current object
func (m *InferenceClassificationOverride) Serialize(writer i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.SerializationWriter)(error) {
    err := m.Entity.Serialize(writer)
    if err != nil {
        return err
    }
    if m.GetClassifyAs() != nil {
        cast := (*m.GetClassifyAs()).String()
        err = writer.WriteStringValue("classifyAs", &cast)
        if err != nil {
            return err
        }
    }
    {
        err = writer.WriteObjectValue("senderEmailAddress", m.GetSenderEmailAddress())
        if err != nil {
            return err
        }
    }
    return nil
}
// SetClassifyAs sets the classifyAs property value. The classifyAs property
func (m *InferenceClassificationOverride) SetClassifyAs(value *InferenceClassificationType)() {
    m.classifyAs = value
}
// SetSenderEmailAddress sets the senderEmailAddress property value. The senderEmailAddress property
func (m *InferenceClassificationOverride) SetSenderEmailAddress(value EmailAddressable)() {
    m.senderEmailAddress = value
}

File: msgraph-mail/ruby/graphrubyv4/models/inference_classification_override.rb
require 'microsoft_kiota_abstractions'
require_relative './entity'
require_relative './models'

module Graphrubyv4::Models
    class InferenceClassificationOverride < Entity
        include MicrosoftKiotaAbstractions::Parsable
        ## 
        # The classifyAs property
        @classify_as
        ## 
        # The senderEmailAddress property
        @sender_email_address
        ## 
        ## Gets the classifyAs property value. The classifyAs property
        ## @return a inference_classification_type
        ## 
        def classify_as
            return @classify_as
        end
        ## 
        ## Sets the classifyAs property value. The classifyAs property
        ## @param value Value to set for the classifyAs property.
        ## @return a void
        ## 
        def classify_as=(classifyAs)
            @classify_as = classifyAs
        end
        ## 
        ## Instantiates a new InferenceClassificationOverride and sets the default values.
        ## @return a void
        ## 
        def initialize() 
            super
        end
        ## 
        ## Creates a new instance of the appropriate class based on discriminator value
        ## @param parseNode The parse node to use to read the discriminator value and create the object
        ## @return a inference_classification_override
        ## 
        def create_from_discriminator_value(parse_node) 
            return nil;
        end
        ## 
        ## The deserialization information for the current model
        ## @return a i_dictionary
        ## 
        def get_field_deserializers() 
            return super.merge({
                "classifyAs" => lambda {|n| @classify_as = n.get_enum_value(Graphrubyv4::Models::InferenceClassificationType) },
                "senderEmailAddress" => lambda {|n| @sender_email_address = n.get_object_value(Graphrubyv4::Models::EmailAddress) },
            })
        end
        ## 
        ## Gets the senderEmailAddress property value. The senderEmailAddress property
        ## @return a email_address
        ## 
        def sender_email_address
            return @sender_email_address
        end
        ## 
        ## Sets the senderEmailAddress property value. The senderEmailAddress property
        ## @param value Value to set for the senderEmailAddress property.
        ## @return a void
        ## 
        def sender_email_address=(senderEmailAddress)
            @sender_email_address = senderEmailAddress
        end
        ## 
        ## Serializes information the current object
        ## @param writer Serialization writer to use to serialize this model
        ## @return a void
        ## 
        def serialize(writer) 
            super
            writer.write_enum_value("classifyAs", @classify_as)
            writer.write_object_value("senderEmailAddress", @sender_email_address)
        end
    end
end
